templates/deployment.yaml

```diff
@@ -27,12 +27,22 @@ spec:
 {{ toYaml .Values.podLabels | indent 8 }}
         {{- end }}
       annotations:
+{{- if .Values.envRenderSecret }}
+        checksum/secret-env: {{ include (print $.Template.BasePath "/secret-env.yaml") . | sha256sum }}
+{{- end }}
         checksum/config: {{ include (print $.Template.BasePath "/configmap.yaml") . | sha256sum }}
-{{- if $.Values.podAnnotations }}
-{{ toYaml $.Values.podAnnotations | indent 8 }}
+{{- if .Values.podAnnotations }}
+{{ toYaml .Values.podAnnotations | indent 8 }}
 {{- end }}
     spec:
-      {{- if .Values.imagePullSecrets }}
+    {{- if .Values.serviceAccount.create }}
+    {{- if .Values.serviceAccount.name }}
+      serviceAccountName: {{ .Values.serviceAccount.name }}
+    {{- else }}
+      serviceAccountName: {{ include "docker-registry.fullname" . }}
+    {{- end }}
+    {{- end }}
```

Comment:
When `serviceAccount.create` is false, but `serviceAccount.name` is set, we still want to set `serviceAccountName` here. Only if `create: false` and `name` is empty/unset do we want to abstain from setting `serviceAccountName` here.

Reply:
This comment is still relevant and has not been addressed.

Reply:
Can you explain how you addressed this feedback? The discussion was marked as resolved, but the code does not appear to have changed and there is no reply to my comment.

Reply:
@ichasco-heytrade, will you be able to update this PR shortly?